src/test/rpc/Simulate_test.cpp

```diff
@@ -0,0 +1,989 @@
+            BEAST_EXPECT(
+                resp[jss::result][jss::error_message] ==
+                "Transaction should not be signed.");
+        }
+    }
+    void
+    testFeeError()
+    {
+        testcase("Fee failure");
+
+        using namespace jtx;
+
+        Env env(*this, envconfig([](std::unique_ptr<Config> cfg) {
+            cfg->section("transaction_queue")
+                .set("minimum_txn_in_ledger_standalone", "3");
+            return cfg;
+        }));
+
+        Account const alice{"alice"};
+        env.fund(XRP(1000000), alice);
+        env.close();
+
+        // fill queue
+        auto metrics = env.app().getTxQ().getMetrics(*env.current());
+        for (int i = metrics.txInLedger; i <= metrics.txPerLedger; ++i)
+            env(noop(alice));
+
+        {
+            Json::Value params;
+            params[jss::tx_json] = noop(alice);
+
+            auto const resp = env.rpc("json", "simulate", to_string(params));
+            auto const result = resp[jss::result];
+            if (BEAST_EXPECT(result.isMember(jss::error)))
+            {
+                BEAST_EXPECT(result[jss::error] == "highFee");
+                BEAST_EXPECT(result[jss::error_code] == rpcHIGH_FEE);
+                BEAST_EXPECT(
+                    result[jss::error_message] ==
+                    "Fee of 8889 exceeds the requested tx limit of 100");
+            }
+        }
+    }
+    void
+    testSuccessfulTransaction()
+    {
+        testcase("Successful transaction");
+
+        using namespace jtx;
+        Env env(*this);
+        static auto const newDomain = "123ABC";
+
+        {
+            auto validateOutput = [&](Json::Value const& resp,
+                                      Json::Value const& tx) {
+                auto result = resp[jss::result];
+                checkBasicReturnValidity(
+                    result, tx, 1, env.current()->fees().base);
+
+                BEAST_EXPECT(result[jss::engine_result] == "tesSUCCESS");
+                BEAST_EXPECT(result[jss::engine_result_code] == 0);
+                BEAST_EXPECT(
+                    result[jss::engine_result_message] ==
+                    "The simulated transaction would have been applied.");
+
+                if (BEAST_EXPECT(
+                        result.isMember(jss::meta) ||
+                        result.isMember(jss::meta_blob)))
+                {
+                    Json::Value metadata;
+                    if (result.isMember(jss::meta_blob))
+                    {
+                        auto unHexed =
+                            strUnHex(result[jss::meta_blob].asString());
+                        SerialIter sitTrans(makeSlice(*unHexed));
+                        metadata = STObject(std::ref(sitTrans), sfGeneric)
+                                       .getJson(JsonOptions::none);
+                    }
+                    else
+                    {
+                        metadata = result[jss::meta];
+                    }
```

Comment:
Please add  new line

Reply:
Fixed in [8f65672](https://github.com/XRPLF/rippled/pull/5069/commits/8f65672adbc7ddc03fc4cb8b6350526f035a0b49)